### ISSUE TITLE
Hidden OS on GPT/EFI

### ISSUE THREAD

**Issue by saraoem** [veracrypt/VeraCrypt#831]
# Current behavior

"System - Create Hidden OS" option is disabled (greyed out). No explanation. Nothing mentioned in the [limitations](https://www.veracrypt.fr/en/Issues%20and%20Limitations.html) page.
Plenty of people have been struggling with this problem before, often without getting any answer at all:
https://sourceforge.net/p/veracrypt/discussion/general/thread/9918b71b/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/85d0e3774d/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/3feb2d46/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/f90bcf05/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/fc02b6b39a/
https://sourceforge.net/p/veracrypt/discussion/general/thread/e69625d993/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/829fcf2c24/
https://sourceforge.net/p/veracrypt/discussion/general/thread/3d7ce769e6/
https://sourceforge.net/p/veracrypt/discussion/general/thread/284f480aa9/
https://sourceforge.net/p/veracrypt/discussion/technical/thread/aaeeb26b/
I see that the thing blocking it for most is the GPT formatted disk. However, newer systems are often coming without a Legacy option, and EFI makes GPT the only option for booting Win10. At least for average user.

# Desired behavior

1) it would be nice to have an explanation instead of just a greyed out button. Maybe a notice about GPT on mouseover? There are probably other things that cause it to be disabled, but still...
2) obviously it would be best to have this option (creating Hidden OS while on EFI/GPT) available. I understand that this is no small feat to achieve, but with Windows 11 upon us, it's even more pressing than before.

# Your Environment

**VeraCrypt version:** 1.24-Update7

**Operating system and version:** Windows 10 build 19043

**Comment by yangu-hury** [comment]
I agree. The development of a new version with support to GPT is very important. I did read that they are already working on it, but it wasn't included in veracrypt project until now. The feature need many improvements before release.

https://sourceforge.net/p/veracrypt/discussion/technical/thread/f90bcf05/

**Comment by Coki91** [comment]
I agree that this problem is increasingly important as legacy hardware is just phased out, at this point where GPT/UEFI are kind of mandatory on not Unsupported systems (besides Linux) it feels wrong to say VeraCrypt supports such feature

I'm trying to come up with bandaid solutions to try later, brainstorming, or better any maintaner statement on the matter would be appreciated

**Comment by metorm** [comment]
I agree
